pytorch_blade/pytorch_blade/torch-mlir/lib/Conversion/TorchToMhlo/DiscTorchToMhlo.cpp

```diff
@@ -189,24 +189,10 @@ class ConvertAtenExtractOp : public OpConversionPattern<AtenOpT> {
     auto elemTy = inpTy.getElementType();
 
     if (outTy != elemTy) {
-      auto output =
-          rewriter.create<tensor::ExtractOp>(op.getLoc(), elemTy, input);
-
-      bool toWider =
-          outTy.getIntOrFloatBitWidth() > elemTy.getIntOrFloatBitWidth();
-      if (elemTy.isIntOrIndex()) {
-        if (toWider) {
-          rewriter.replaceOpWithNewOp<arith::ExtSIOp>(op, outTy, output);
-        } else {
-          rewriter.replaceOpWithNewOp<arith::TruncIOp>(op, outTy, output);
-        }
-      } else {
-        if (toWider) {
-          rewriter.replaceOpWithNewOp<arith::ExtFOp>(op, outTy, output);
-        } else {
-          rewriter.replaceOpWithNewOp<arith::TruncFOp>(op, outTy, output);
-        }
-      }
+      // outTy and elemTy may be diffrent datatypes like int to float
```

Comment:
I think `mhlo::ConvertOp` may introduce much redundant data access here. How about extracting the element first and then handling the type conversion/casting carefully as before？